orderer/common/filerepo/filerepo_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo_test
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/orderer/common/filerepo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFileRepo_Save(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	t.Run("Successful save", func(t *testing.T) {
+		err := r.Save("newchannel", []byte("block-bytes"))
+		defer os.Remove("testdata/newchannel.joinblock")
+		assert.NoError(t, err)
+
+		// Check file bytes
+		bytes, err := ioutil.ReadFile("testdata/newchannel.joinblock")
+		require.NoError(t, err)
+		assert.Equal(t, bytes, []byte("block-bytes"))
+
+		// Check tempfile doesn't exist
+		_, err = os.Stat("testdata/~newchannel")
+		assert.True(t, os.IsNotExist(err))
+	})
+
+	t.Run("Failed save, invalid channel name", func(t *testing.T) {
+		err := r.Save("InvalidChannelName", []byte{})
+		assert.EqualError(t, err, "'InvalidChannelName' contains illegal characters")
+	})
+}
+
+func TestFileRepo_Scan(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	t.Run("Successful scan", func(t *testing.T) {
+		files, err := r.Scan()
+		assert.NoError(t, err)
+		assert.Equal(t, files, []string{"mychannel.joinblock"})
+	})
+
+	t.Run("Failed scan, invalid dir", func(t *testing.T) {
+		r.Options.FileRepoDir = "invaliddir"
+
+		_, err := r.Scan()
+		assert.EqualError(t, err, "open invaliddir: no such file or directory")
+	})
+}
+
+func TestFileRepo_Load(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	t.Run("Successful load", func(t *testing.T) {
+		bytes, err := r.Load("testdata/mychannel.joinblock")
+		assert.NoError(t, err)
+		block := &common.Block{}
+		err = proto.Unmarshal(bytes, block)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Failed load, invalid file", func(t *testing.T) {
+		_, err := r.Load("invalidfile")
+		assert.EqualError(t, err, "open invalidfile: no such file or directory")
+	})
+}
+
+func TestFileRepo_FileToChannelName(t *testing.T) {
+	options := filerepo.Options{
+		FileSuffix:          ".joinblock",
+		TransientFilePrefix: "~",
+		FileRepoDir:         "testdata",
+	}
+
+	r := filerepo.Repo{
+		Options: options,
+	}
+
+	channelName := r.FileToChannelName("testdata/mychannel.joinblock")
+	assert.Equal(t, channelName, "mychannel")
```

Comment:
Shouldn't the order of argument be `expected` and then `actual` value?

Reply:
I'm so used to gomega matchers which is the opposite 😅 having the actual come first.